flatten_json.py

```diff
@@ -16,7 +16,7 @@ def _construct_key(previous_key, separator, new_key):
     :return: a string if previous_key exists and simply passes through the new_key otherwise
     """
     if previous_key:
-        return "{}{}{}".format(previous_key, separator, new_key)
+        return u"{}{}{}".format(previous_key, separator, new_key)
```

Comment:
Do we always want to do this? Does it make sense to `try` the original version and do this in the case of failure?

Reply:
I haven't dealt with these issues much. Based on this [post](https://stackoverflow.com/questions/18034272/python-str-vs-unicode-types) It sounds like using `unicode` is the better approach but not sure if there are any unintended consequences?!

Reply:
If we use unicode it fails for python3, as the type name has changed. That's why you will see another commit from my side. Actually instead of a try...except I added an if clause to check the type and only do the conversion when needed. This was a solution that works for every case, but it returns always unicode. The best part is that it works in any possible scenario, but people will need to compare names properly :-)

Reply:
Also, we built this because in PYBOSSA we have people using it from Greece, China, India, etc. so we've always a mix of strings and UTF8 chars. That's how we discovered this issue. Right now with our patch it works without problems ;-)